api/lib_test.go

```diff
@@ -70,6 +70,45 @@ func TestCreateFailsWithBadData(t *testing.T) {
 	require.Error(t, err)
 }
 
+func TestPin(t *testing.T) {
```

Comment:
Is there an error case (besides filesystem corruption) we could test?

How about querying for a different checksum that was not registered and ensuring this provides a proper error message, not a panic.